### ISSUE TITLE
Chr() function does not support Chinese characters?

### ISSUE THREAD

**Issue by dj-make** [twinbasic/twinbasic#1457]
Chr() function does not support Chinese characters?
![QQ截图20230113041807](https://user-images.githubusercontent.com/122567615/212174068-bbd0c22d-b289-46fe-bf7d-b2e85f26c80d.jpg)
![QQ截图20230113041837](https://user-images.githubusercontent.com/122567615/212174074-bf0d501d-46d9-407d-9b2d-3750ebf778c5.jpg)
![QQ截图20230113041914](https://user-images.githubusercontent.com/122567615/212174077-96167b19-c274-4577-9863-03446cc62146.jpg)

**Comment by bclothier** [comment]
I'm pretty sure it does not, not even in VBx. I think you should be using `ChrW()`.

**Comment by mburns08109** [comment]
yeah, in terms of 100% back-compat - that sounds exactly right.

**Comment by dj-make** [comment]
> 我很确定它没有，即使在 VBx 中也是如此。我认为你应该使用.`ChrW()`

The result shows garbled code
![QQ截图20230113045017](https://user-images.githubusercontent.com/122567615/212178723-6250dc0a-198a-4e4c-8224-b0abffd3933e.jpg)

**Comment by wqweto** [comment]
@bclothier `ChrW` is different (Unicode).

`Chr` must support "multi-byte ANSI" on certain locales where (chinese) chars are as wide as an `Integer` not only limited to single `Byte`s.

As a consequence you get some possible negative values as well which is ugly as hell. I've never user nor seen this multi-byte "story" but it looks scary so I'll leave Wayne figure it out :-))

**Comment by WaynePhillipsEA** [comment]
Indeed, there will likely be several DBCS issues that will need looking at, as our audience grows :)

**Comment by bclothier** [comment]
TBH I completely forgot about the DBCS. I think @wqweto  is right about it being scary because if I understand (and I do not speak from direct experience so this is hearsay), it's possible to have one project that works well in one locale only to come out garbled if it's loaded on another computer using a different locale. 

One idea is to have a project level setting that default to Windows' local but can be changed to be used in the import/export process and used to influence the behavior of `Chr()` or other functions that depends on locale. IINM, the underlying functions do accept a LCID parameters but do not expose it via the VBx. That would be a huge improvement for migrating the legacy VBx codebase while making it easy to be localized for different language without having to change the original code. To make it a zero-code-change, manifest files could be used, perhaps.
